core/src/replay_stage.rs

```diff
@@ -1461,6 +1530,116 @@ pub(crate) mod tests {
         );
     }
 
+    fn new_replay_stage_for_testing(
```

Comment:
@TristanDebrunner we've been trying to avoid large test harnesses like this in unit tests. Ideally you extract out the critical sections of the code  away from all the externalities like PohService, BankForks, etc. and just test persist/restore.

Then to test integration level behavior, write a test in local_cluster and see that we can properly restart a node with the correct tower/check the tower state to make sure its correct.

Reply:
So no unit tests with an actual `ReplayStage`? This was the minimum I could find to instantiate a `ReplayStage` to test with. These tests are still fast (much faster than a `LocalCluster` test)